hibernate-core/src/main/java/org/hibernate/type/EntityType.java

```diff
@@ -681,7 +681,7 @@ protected final Object resolveIdentifier(Serializable id, SharedSessionContractI
 				getAssociatedEntityName(),
 				id,
 				eager,
-				isNullable() && !isProxyUnwrapEnabled
+				isNullable()
 		);
 
```

Comment:
@sebersole, is proxy unwrapping still supported w/o bytecode enhancement?

Reply:
proxy unwrapping without bytecode enhancement would just be eager loading.  so in that sense yes and no